Source/WebCore/dom/ScriptExecutionContext.cpp

```diff
@@ -253,14 +253,10 @@ void ScriptExecutionContext::dispatchMessagePortEvents()
 
     auto completionHandlers = std::exchange(m_processMessageWithMessagePortsSoonHandlers, Vector<CompletionHandler<void()>> { });
 
-    // Make a frozen copy of the ports so we can iterate while new ones might be added or destroyed.
-    for (RefPtr messagePort : copyToVectorOf<RefPtr<MessagePort>>(m_messagePorts)) {
-        // The port may be destroyed, and another one created at the same address,
-        // but this is harmless. The worst that can happen as a result is that
-        // dispatchMessages() will be called needlessly.
-        if (m_messagePorts.contains(messagePort.get()) && messagePort->started())
-            messagePort->dispatchMessages();
-    }
+    m_messagePorts.forEach([](auto& messagePort) {
+        if (messagePort.started())
+            messagePort.dispatchMessages();
+    });
```

Comment:
What makes it now safe to iterate this while ports might be added or removed?

Reply:
To be clear, `for (auto& t : weakHashSet)` doesn't make a copy but `WeakHashSet::forEach()` does and I think this is why we have this function. To your point, the name doesn't make it clear. I'd be happy to rename this function in a separate PR if we find a good name. Maybe `WeakHashSet::safeIterate()`, `WeakHashSet::iterateOnCopy()`, `WeakHashSet::safeForEach()`